### PR TITLE
Adds octopus animal

### PR DESCRIPTION
**Pull request by JasonMFry**
Let me know if there's anything I should change. Addresses https://github.com/AllureOfTheStars/Allure/issues/84

### CODE REVIEW

**Comment by Mikolaj** [comment]
Wow, well done! Thank you. Let me nitpick directly in the code. 

BTW, appveyor CI run of your PR will fail --- something broke there yesterday; perhaps they just made the machines slower and I go over the limit. Don't bother.

File: GameDefinition/Content/ItemKindActor.hs [hunk 1]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
                , ("eye 6", COrgan), ("ear 9", COrgan)
                , ("animal brain", COrgan) ]
   }
+octopus = ItemKind
+  { isymbol  = 'o'
+  , iname    = "Giant octopus"
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: Would you like it to be a unique monster (at most one occurrence in the game)? The convention is, for uniques, we capitalize all, for non-uniques, we use the capitalization from Wikipedia.

File: GameDefinition/Content/ItemKindActor.hs [hunk 2]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
...
+  , ifreq    = [ ("animal", 100), ("mobile", 1), ("mobile animal", 100) ]
+  , iflavour = zipPlain [BrMagenta]
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: Well done. Pink is not overused, matches the Wikipedia photo, and it so happens that pink actors have ranged attacks, so far.

File: GameDefinition/Content/ItemKindActor.hs [hunk 3]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
...
+  , icount   = 1
+  , irarity  = [(1, 5)]
+  , iverbHit = "thud"
+  , iweight  = 15000
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: They are said to weight up to 70kg, but probably most of the mass is in the tentacles, so 15kg for the trunk makes sense. OTOH, if you wish, you can exaggerate a bit and make our own octopus a sight to behold. :) OTOH, if you go for an exceptionally large specimen, you may want to bump HP, given that it doesn't have armored skin and so is rather fragile.

File: GameDefinition/Content/ItemKindActor.hs [hunk 4]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
...
+  , idamage  = 0
+  , iaspects = [ AddSkill SkMaxHP 14, AddSkill SkMaxCalm 80
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: Perfect. And I'd add this:

https://github.com/AllureOfTheStars/Allure/blob/74ecdf787d3cbffca8a80c8ac6f5d20f187c941f/GameDefinition/Content/ItemKindActor.hs#L468

File: GameDefinition/Content/ItemKindActor.hs [hunk 5]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
...
+               , AddSkill SkSpeed 27, AddSkill SkNocto 3 -- good night vision
+               , AddSkill SkAlter (-2), SetFlag Durable ]  -- can't use stairs nor doors
+  , ieffects = []
+  , idesc    = "It has eight arms of rage."
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: !!! :D

File: GameDefinition/Content/ItemKindActor.hs [hunk 6]
```diff
@@ -474,6 +474,24 @@ alligator = ItemKind
...
+  , ikit     = [ ("tentacle 8", COrgan), ("small claw", COrgan)
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: Well chosen organs, in particular sight --- 2 tiles better than human. As you can see here

https://github.com/AllureOfTheStars/Allure/blob/74ecdf787d3cbffca8a80c8ac6f5d20f187c941f/GameDefinition/Content/ItemKindOrgan.hs#L150

tentacles come in bunches of four. So it's enough to say `("tentacle", COrgan), ("tentacle", COrgan)` and we'll get eight of them.

OTOH, the claws come in twos --- is that right? I didn't read enough on octopuses, so I didn't verify if they have claws at the end of tentacles or something similar to mandibles in their mouths (beaks? we'd need to define such a new organ) or wherever else and in what number. They are mildly venomous, so we could add the `venomTooth`, if you so please.

Anyway, here starts the most fun part --- designing the special organ, or two, that makes octopus so different --- the siphon and the ink sack, if I got it right, or both in one. They could be similar to the vents but probably more interesting. We don't need to model all at once, especially if a large engine extension would be required, but we can certainly encode some ideas in the organ name(s), descriptions and comments. Let's brainstorm.

**Comment by Mikolaj** [comment]
First wild brainstorm idea: perhaps whenever octopus attacks with a jet of water, it should be pushed in the opposite direction than the attack (more or less, that would be implemented with the `Pull` effect that moves the actor in the opposite direction than his last move, so often it would match). That would capture both the attack and the speedy fleeing behaviour using in one package.

File: GameDefinition/Content/ItemKindActor.hs [hunk 7]
```diff
@@ -474,7 +474,7 @@ alligator = ItemKind
                , ("eye 6", COrgan), ("ear 9", COrgan)
                , ("animal brain", COrgan) ]
   }
-octopus = ItemKind
+giantOctopus = ItemKind
```
Comment (Mikolaj) [inline GameDefinition/Content/ItemKindActor.hs]: BTW, the identifier needs to be mentioned at the start of the file in two places.

**Comment by Mikolaj** [comment]
BTW, the Allure on Windows crash is now resolved (was text locale mismatch).

File: GameDefinition/Content/ItemKindActor.hs (hunk 1 above)
Comment (JasonMFry) [reply GameDefinition/Content/ItemKindActor.hs]: I thought `Giant octopus` is the capitalization from Wikipedia. But should I change it to `giant octopus`? I don't want it to be unique.

File: GameDefinition/Content/ItemKindActor.hs (hunk 3 above)
Comment (JasonMFry) [reply GameDefinition/Content/ItemKindActor.hs]: The record is 71kg, so I'm going to make it 72kg ;)

File: GameDefinition/Content/ItemKindActor.hs (hunk 6 above)
Comment (JasonMFry) [reply GameDefinition/Content/ItemKindActor.hs]: I'm thinking the octopus should have 1) a beak, 2) suckers, and 3) ink sack. I will give a first attempt at an ink sack later.

File: GameDefinition/Content/ItemKindActor.hs (hunk 1 above)
Comment (Mikolaj) [reply GameDefinition/Content/ItemKindActor.hs]: Indeed, they are not consistent in capitalization. Let's pick `giant octopus`; I think it's the more common one.

File: GameDefinition/Content/ItemKindActor.hs (hunk 3 above)
Comment (Mikolaj) [reply GameDefinition/Content/ItemKindActor.hs]: Perfect. I guess the lower gravity servers them well. :)

File: GameDefinition/Content/ItemKindActor.hs (hunk 6 above)
Comment (Mikolaj) [reply GameDefinition/Content/ItemKindActor.hs]: Go for it! 

Will the ink sac be the same as the siphon? No pressure to add both, it's fine to omit organs --- the ones we don't mention are considered to be a part of the trunk and mostly passive or to cooperate with the organs we do mention.

**Comment by Mikolaj** [comment]
While you are finish this up (no hurry, it's meant to be fun and the release is still far away :) ), I've added a mechanism that spawns aquatic actors peferably in water and aquatic animals on some water-rich levels, so you may want ot add `("aquatic", 1), ("aquatic animal", 90)` to the `ifreq` field.

**Comment by Mikolaj** [comment]
@JasonMFry: I'd like to test actor generation, in particular in water and on water-rich levels for the upcoming release, so if you do not mind, I'd merge-rebase this PR and we can open an issue about the ink sac organ. Github is not particularly well suited to long-running PRs anyway, so we can focus on the technicalities of the new organ in a new PR.

**Comment by Mikolaj** [comment]
@JasonMFry: thanks a lot! Future work moved to #91.

**Pull request merged**